apps/mobile/src/features/send-refactor/forms/stx/stx-form.tsx

```diff
@@ -0,0 +1,159 @@
+import { Controller } from 'react-hook-form';
+
+import { AmountField } from '@/features/send-refactor/components/amount-field';
+import { AssetDisplay } from '@/features/send-refactor/components/asset-display';
+import { ErrorMessage } from '@/features/send-refactor/components/error-message';
+import { Memo } from '@/features/send-refactor/components/memo';
+import { Numpad } from '@/features/send-refactor/components/numpad';
+import { Recipient } from '@/features/send-refactor/components/recipient';
+import {
+  SendFormContainer,
+  SendFormFooter,
+} from '@/features/send-refactor/components/send-form-layout';
+import { useStxForm } from '@/features/send-refactor/forms/stx/use-stx-form';
+import { useSendFlowContext } from '@/features/send-refactor/send-flow-provider';
+import { locale } from '@/features/send-refactor/temporary-constants';
+import { Account } from '@/store/accounts/accounts';
+import { t } from '@lingui/macro';
+
+import { FiatCurrency, MarketData, Money } from '@leather.io/models';
+import { Button, StxAvatarIcon } from '@leather.io/ui/native';
+import { isNumber } from '@leather.io/utils';
+
+const protocol = 'nativeStx';
+const symbol = 'STX';
+
+interface StxFormProps {
+  account: Account;
+  availableBalance: Money;
+  fiatBalance: Money;
+  nonce: number | undefined;
+  marketData: MarketData;
+  fiatCurrency: FiatCurrency;
+  assetItemAnimationOffsetTop?: number | null;
+  onOpenAssetPicker(): void;
+}
+
+export function StxForm({
+  assetItemAnimationOffsetTop,
+  availableBalance,
+  fiatBalance,
+  fiatCurrency,
+  marketData,
+  nonce,
+  onOpenAssetPicker,
+  account,
+}: StxFormProps) {
+  const {
+    state: { currencyMode },
+  } = useSendFlowContext();
+  const currency = currencyMode === 'crypto' ? symbol : fiatCurrency;
```

Comment:
This code seems to get repeated a lot
```
const currency = currencyMode === 'crypto' ? symbol : fiatCurrency;
```

Maybe it's worthy of a lookup helper?

Reply:
Hate this wholeheartedly. Working on it already.

Reply:
Addressed [here](https://github.com/leather-io/mono/pull/984/files/fd83e6cc86c5f4e8f396de1bd34cca46fe528665#r2009694420)